File: windows/DMShot.Tests/UpdateHintTests.cs
using DMShot.Update;
using Xunit;

public class UpdateHintTests
{
    [Fact]
    public void AvailableAndReadyYieldTheirVersion()
    {
        Assert.Equal("1.2.3", UpdateHint.VersionFor(UpdateState.ForAvailable("1.2.3", new List<ChangelogVersion>())));
        Assert.Equal("2.0.0", UpdateHint.VersionFor(UpdateState.ForReadyToInstall("2.0.0")));
    }

    [Fact]
    public void AllOtherStatesYieldNoHint()
    {
        var silent = new[]
        {
            UpdateState.Disabled, UpdateState.Idle, UpdateState.Checking, UpdateState.UpToDate,
            UpdateState.ForDownloading(50), UpdateState.ForError("boom"),
        };
        foreach (var state in silent)
            Assert.Null(UpdateHint.VersionFor(state));
    }
}

File: mac/Tests/DMShotTests/UpdateHintTests.swift
import XCTest
@testable import DMShot

final class UpdateHintTests: XCTestCase {
    func testAvailableAndReadyYieldTheirVersion() {
        XCTAssertEqual(UpdateHint.version(for: .available(version: "1.2.3", notes: [])), "1.2.3")
        XCTAssertEqual(UpdateHint.version(for: .readyToInstall(version: "2.0.0")), "2.0.0")
    }

    func testAllOtherStatesYieldNoHint() {
        let silent: [UpdateState] = [
            .disabled, .idle, .checking, .upToDate,
            .downloading(percent: 50), .extracting,
            .error(message: "boom"),
        ]
        for state in silent {
            XCTAssertNil(UpdateHint.version(for: state), "expected no hint for \(state)")
        }
    }
}
